src/promptflow-devkit/promptflow/_sdk/operations/_trace_operations.py

```diff
@@ -81,3 +92,103 @@ def list_line_runs(
             line_run._append_evaluations(eval_line_runs)
             line_runs.append(line_run)
         return line_runs
+
+    @monitor_operation(activity_name="pf.traces.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        run: typing.Optional[str] = None,
+        collection: typing.Optional[str] = None,
+        started_before: typing.Optional[typing.Union[str, datetime.datetime]] = None,
+    ) -> None:
+        """Delete traces permanently.
+
+        Support delete according to:
+          - run
+          - non default collection
+          - collection combined with time as started before
+
+        Examples:
+          - pf.traces.delete(run="name")
+          - pf.traces.delete(collection="collection")
+          - pf.traces.delete(collection="default", started_before="2024-03-19T15:17:23.807563")
+
+        :param run: Name of the run.
+        :type run: Optional[str]
+        :param session: Id of the session.
+        :type session: Optional[str]
+        :param started_before: ISO 8601 format time string (e.g., "2024-03-19T15:17:23.807563").
+        :type started_before: Optional[Union[str, datetime.datetime]]
+        """
+        self._logger.debug(
+            "delete traces with parameters, run: %s, collection: %s, started_before: %s",
+            run,
+            collection,
+            started_before,
+        )
+        self._validate_delete_query_params(run=run, collection=collection, started_before=started_before)
+        self._logger.debug("try to delete line run(s)...")
+        if isinstance(started_before, str):
+            started_before = datetime.datetime.fromisoformat(started_before)
+        self._delete_within_transaction(run=run, collection=collection, started_before=started_before)
+
+    def _validate_delete_query_params(
+        self,
+        run: typing.Optional[str] = None,
+        collection: typing.Optional[str] = None,
+        started_before: typing.Optional[typing.Union[str, datetime.datetime]] = None,
+    ) -> None:
+        # valid delete queries:
+        #   1. run=xxx
+        #   2. collection=yyy
+        #   3. collection=zz, started_before=zz
+        # this function will directly return for valid cases
+        if run is not None and collection is None and started_before is None:
+            return
+        if collection is not None and run is None:
+            if started_before is not None:
+                # if `started_before` is a time string, need to ensure it's in valid ISO 8601 format
+                if isinstance(started_before, str):
+                    try:
+                        datetime.datetime.fromisoformat(started_before)
+                        return
+                    except ValueError:
```

Comment:
why pass?

Reply:
i played some tricks in this validate function: at the very end, it will raise an user error; so for such error, just pass, and let the code go to the end of this function.